x-pack/solutions/search/plugins/search_query_rules/public/components/query_ruleset_detail/query_rule_flyout/document_selector/draggable_list.tsx

```diff
@@ -35,70 +36,97 @@ export const DraggableList: React.FC<DraggableListProps> = ({
-                data-test-subj={`queryRuleDocumentDraggable-id-${doc}-${index}`}
-              >
-                {() => (
-                  <DocumentSelector
-                    initialDocId={doc}
-                    indexDoc={index}
-                    type={pinType}
-                    hasIndexSelector={false}
-                    onDeleteDocument={() => {
-                      onDeleteDocument(index);
-                    }}
-                    onIdSelectorChange={(id) => {
-                      onIdSelectorChange(index, id);
-                    }}
-                  />
-                )}
-              </EuiDraggable>
-            ))
-          : actionFields.map((doc, index) => (
-              <EuiDraggable
-                usePortal
-                spacing="m"
-                index={index}
-                hasInteractiveChildren={true}
-                draggableId={'queryRuleDocumentDraggable-' + doc._id + '-' + index}
-                key={doc._id}
-                isDragDisabled={pinType === 'exclude'}
-                data-test-subj={`queryRuleDocumentDraggable-doc-${doc._id}-${index}`}
-              >
-                {() => (
-                  <DocumentSelector
-                    initialDocId={doc._id}
-                    indexDoc={index}
-                    type={pinType}
-                    index={doc._index}
-                    onDeleteDocument={() => {
-                      onDeleteDocument(index);
-                    }}
-                    onIndexSelectorChange={(indexName) => {
-                      onIndexSelectorChange(index, indexName);
+    <>
+      <EuiDragDropContext onDragEnd={dragEndHandle}>
+        <EuiDroppable droppableId="queryRuleDroppable" spacing="m">
+          <>
+            {actionFields.length === 0 && actionIdsFields?.length === 0 && (
+              <EuiCallOut
+                title="At least one document is required"
+                color="warning"
+                size="s"
+                iconType="iInCircle"
+              />
+            )}
+            {isIdRule && actionIdsFields
+              ? actionIdsFields.map((doc, index) => (
+                  <EuiDraggable
+                    usePortal
+                    spacing="m"
+                    index={index}
+                    hasInteractiveChildren={true}
+                    draggableId={'queryRuleDocumentDraggable-' + doc + '-' + index}
+                    key={doc + '-' + index}
+                    isDragDisabled={pinType === 'exclude'}
+                    data-test-subj={`queryRuleDocumentDraggable-id-${doc}-${index}`}
+                  >
+                    {() => {
+                      const docError = errors?.actions?.docs?.[index];
+                      // check if it can be cast into FieldError from React Hook Form and cast it otherwise undefined
+                      const fieldError =
+                        docError?.type && docError.message ? (docError as FieldError) : undefined;
```

Comment:
a `isFieldError` function would be nice here so that we didn't need to cast. Something like:

`function isFieldError(error: {type?: ___; message?: string; }): error is FieldError`